pages/api/subscribe.ts

```diff
@@ -0,0 +1,127 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import md5 from "md5";
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any
+export default async function handler(req: NextApiRequest, res: NextApiResponse): Promise<any> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const returnData: any = {};
+  const mergeFields = { ...req.body, "CONSENT": String(req.body.CONSENT) };
+  console.log(`mergeFields: ${mergeFields}`);
+
+  const checkSubscriber = await fetch(
+    `https://us5.api.mailchimp.com/3.0/lists/${process.env.MAILCHIMP_LIST_ID}/members/${md5(mergeFields.EMAIL)}`,
+    {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `auth ${process.env.MAILCHIMP_API_KEY}-us5`,
+      },
+    }
+  );
+
+  const checkSubscriberData = await checkSubscriber.json();
+
+  if (checkSubscriberData.status ===  'subscribed') {
+    console.log('child is subscribed');
+    const updateHackathonField = await fetch(`https://us5.api.mailchimp.com/3.0/lists/${process.env.MAILCHIMP_LIST_ID}/members/${md5(mergeFields.EMAIL)}`, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+        "Authorization": `auth ${process.env.MAILCHIMP_API_KEY}-us5`,
+        "Access-Control-Allow-Origin": "*",
+      },
+      body: JSON.stringify({
+        merge_fields: {
+          merge_fields: mergeFields,
```

Comment:
This will overdo the existing merge fields of the child and later the parent, I think that's okay since if the user already exists they may want to update their data.